### DISCUSSION TITLE
Efficiently monitor flux processes

### DISCUSSION THREAD

**Discussion by andreipantelimon** [fluxcd/flux2#3751] [Q&A]
Hey,
How can we monitor the flux processes and have alerts with all the information that we need?
As we are going into production with a Flux integration on our infrastructure, we need to set-up an alerting service which should fire when failed reconciliations are detected in the cluster. We prefer the flexibility that Prometheus/Grafana stack offers us.
The problem is getting the correct error messages of the failed reconciliations, since Flux only exposes `gotk_reconcile_condition`, `gotk_suspend_status` and `gotk_reconcile_duration_seconds_*`. The log-based metric system is not an option since objects are reconciling at different intervals.
Are we missing something?
Thanks

**Comment by makkes** [comment]
If the exposed metrics don't work for your setup then you might want to take a look at [Flux notifications](https://fluxcd.io/flux/components/notification/) and in particular the [Prometheus Alertmanager provider](https://fluxcd.io/flux/components/notification/provider/#prometheus-alertmanager).

**Comment by kingdonb** [comment]
I would suggest you take a look at the AlertManager example under Metrics, in the Flux monitoring guide:

https://fluxcd.io/flux/guides/monitoring/#metrics

It is entirely possible that you missed this. It fires when Flux resources aren't reconciling for longer than 10m. It is OK to use with Flux resources that have a longer interval, because it works based on `Ready`, and those resources which just don't reconcile every 10m it isn't a problem for them, because they are still `Ready`.

I have all sorts of intervals, up to 12h and beyond with no problems, this config tells me when something stopped reconciling with an alert, and it's not based on any messages or filter regex that can be wrong. Hope this is exactly what you were looking for.

Let us know if this could have been more visible, maybe? Thanks for using Flux!
